src/EditorFeatures/Core/IntelliSense/ImportCompletionCacheService/EditorTypeImportCompletionCacheServiceFactory.cs

```diff
@@ -0,0 +1,26 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Composition;
+using Microsoft.CodeAnalysis.Completion.Providers;
+using Microsoft.CodeAnalysis.Editor.Shared.Utilities;
+using Microsoft.CodeAnalysis.Host.Mef;
+
+namespace Microsoft.CodeAnalysis.IntelliSense
+{
+    // This is the implementation at Editor layer to provide a CancellationToken
+    // for the workqueue used for background cache refresh.
+    [ExportWorkspaceServiceFactory(typeof(IImportCompletionCacheService<TypeImportCompletionCacheEntry, TypeImportCompletionCacheEntry>), ServiceLayer.Editor), Shared]
+    internal sealed class EditorTypeImportCompletionCacheServiceFactory
+        : AbstractImportCompletionCacheServiceFactory<TypeImportCompletionCacheEntry, TypeImportCompletionCacheEntry>
+    {
+        [ImportingConstructor]
+        [Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
+        public EditorTypeImportCompletionCacheServiceFactory(IThreadingContext threadingContext)
+            : base(threadingContext.DisposalToken)
```

Comment:
so thsi works.  however, what i was somewhat thinking of was that the ITHreadingContext and IAsyncOperationListener would live here, and the abstract class would expose an abstrac method asking to get the AsyncBatchingWorkQueue to use.    but i don't feel super strongly.  it will help out some later code.